### PR TITLE
Bug 1189519 - Add get platforms endpoint

### PR DESCRIPTION
**Pull request by MikeLing**
[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/848)

### CODE REVIEW

**Comment by MikeLing** [comment]
screen shot 2015-08-08 at 11 33 52 am
Test on OS X 10.10
Mozilla Firefox 39.0

File: treeherder/model/derived/jobs.py [hunk 1]
```diff
@@ -431,6 +431,20 @@ def get_performance_series_summary(self, interval_seconds):
 
         return series_summary
 
+    def get_platforms(self, interval_seconds):
```
Comment (wlach) [inline treeherder/model/derived/jobs.py]: get_performance_platforms would be a better name, since we're getting platforms for performance data here specifically.

File: treeherder/model/derived/jobs.py [hunk 2]
```diff
@@ -431,6 +431,20 @@ def get_performance_series_summary(self, interval_seconds):
...
+            proc="jobs.selects.get_perf_series_properties",
+            placeholders=[last_updated_limit, interval_seconds],
+            debug_show=self.DEBUG,
+        )
+        platforms = []
+        for datum in data:
+            if datum['property'] == 'machine_platform':
+                if datum['value'] not in platforms:
+                    platforms.append(datum['value'])
+        return platforms
```
Comment (wlach) [inline treeherder/model/derived/jobs.py]: You could express this more concisely by using set() and list comprehensions here:

https://docs.python.org/2/library/sets.html
https://docs.python.org/2/tutorial/datastructures.html#list-comprehensions

E.g.:

     platforms = set()
     for [datum for datum in data if datum['property'] == 'machine_platform']:
         platforms.add(datum['value'])

File: treeherder/webapp/api/performance_data.py [hunk 3]
```diff
@@ -110,3 +110,21 @@ def get_performance_data(self, request, project, jm, pk=None):
         data = jm.get_performance_series_from_signatures(signatures, interval_seconds)
 
         return Response(data)
+
+    @list_route()
+    @with_jobs
+    def platforms(self, request, project, jm, pk=None):
+        """
+        Get method implementation for platforms
```
Comment (wlach) [inline treeherder/webapp/api/performance_data.py]: Let's say something more meaningful here, like "Return list of platforms for which we have performance data"

File: treeherder/webapp/api/performance_data.py [hunk 4]
```diff
@@ -110,3 +110,21 @@ def get_performance_data(self, request, project, jm, pk=None):
...
+
+        Input: time interval
+        Output: list of platforms
+        """
+        try:
+            interval_second = int(request.QUERY_PARAMS.get('interval'))
```
Comment (wlach) [inline treeherder/webapp/api/performance_data.py]: As in the other PR, you can just name this "interval".

File: treeherder/model/derived/jobs.py (hunk 1 above)
Comment (wlach) [inline treeherder/model/derived/jobs.py]: I meant this specific method should be renamed to `get_performance_platforms`, not the web service method. :)

So change this and revert the changes there. The behaviour you had before was good (/performance-data/platforms/ is a good endpoint name, as the performance-data part provides enough context to figure out what it's about)

Comment (MikeLing) [inline treeherder/model/derived/jobs.py]: Oh, sorry. I know what's your mean. but somehow change the wrong place. Sorry about that. :(
